internal/services/network/network_interface_resource.go

```diff
@@ -325,6 +355,22 @@ func resourceNetworkInterfaceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		},
 	}
 
+	if d.HasChange("auxiliary_mode") {
+		if auxiliaryMode, hasAuxiliaryMode := d.GetOk("auxiliary_mode"); hasAuxiliaryMode {
+			update.Properties.AuxiliaryMode = pointer.To(networkinterfaces.NetworkInterfaceAuxiliaryMode(auxiliaryMode.(string)))
+		}
+	} else {
+		update.Properties.AuxiliaryMode = existing.Model.Properties.AuxiliaryMode
+	}
```

Comment:
how do users unset this property?

Reply:
If users remove this property from the .tf, `hasAuxiliaryMode = false` so that `update.Properties.AuxiliaryMode` will not be assigned a value, then Azure will unset this property.